runelite-client/src/main/java/net/runelite/client/plugins/poison/PoisonPlugin.java

```diff
@@ -117,21 +141,26 @@ protected void shutDown() throws Exception
 		poisonNaturalCure = null;
 		nextPoisonTick = null;
 		lastValue = -1;
+		lastDiseaseValue = -1;
 
+		clientThread.invoke(this::resetHealthIcon);
 	}
 
 	@Subscribe
 	public void onVarbitChanged(VarbitChanged event)
 	{
 		final int poisonValue = client.getVar(VarPlayer.POISON);
-		if (poisonValue == lastValue)
+		final int diseaseValue = client.getVar(VarPlayer.DISEASE_VALUE);
+		if (poisonValue == lastValue && diseaseValue == lastDiseaseValue)
```

Comment:
This looks wrong

Reply:
If either value changed we want the below code to return so `&&` is the correct operator

Reply:
This is changing how the code worked before, so I think it is incorrect (e.g the poison stuff ran only when poison value was changed, now it runs also if disease value has changed)

Reply:
You're right, this has been updated accordingly